integration-test/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -145,7 +175,64 @@ private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll wit
   }
 
   private def doBasicDriverPodCheck(driverPod: Pod): Unit = {
-    assert(driverPod.getMetadata.getLabels.get("spark-role") === "driver")
+    assert(driverPod.getSpec.getContainers.get(0).getImage === "spark-driver:latest")
+    assert(driverPod.getSpec.getContainers.get(0).getName === "spark-kubernetes-driver")
+  }
+
+  private def doBasicExecutorPodCheck(executorPod: Pod): Unit = {
+    assert(executorPod.getSpec.getContainers.get(0).getImage === "spark-executor:latest")
+    assert(executorPod.getSpec.getContainers.get(0).getName === "executor")
+  }
+
+  private def checkCustomSettings(pod: Pod): Unit = {
+    assert(pod.getMetadata.getLabels.get("label1") === "label1-value")
+    assert(pod.getMetadata.getLabels.get("label2") === "label2-value")
+    assert(pod.getMetadata.getAnnotations.get("annotation1") === "annotation1-value")
+    assert(pod.getMetadata.getAnnotations.get("annotation2") === "annotation2-value")
+
+    val container = pod.getSpec.getContainers.get(0)
+    val envVars = container
+      .getEnv
+      .asScala
+      .map { env =>
+        (env.getName, env.getValue)
+      }
+      .toMap
+    assert(envVars("ENV1") === "VALUE1")
+    assert(envVars("ENV2") === "VALUE2")
+  }
+
+  private def createTestSecret(): Unit = {
+    testBackend.getKubernetesClient.secrets
+      .createNew()
+      .editOrNewMetadata()
+        .withName(TEST_SECRET_NAME)
+        .withNamespace(kubernetesTestComponents.namespace)
+        .endMetadata()
+      .addToStringData(TEST_SECRET_KEY, TEST_SECRET_VALUE)
+      .done()
+  }
+
+  private def checkTestSecret(pod: Pod, withInitContainer: Boolean = false): Unit = {
+    val testSecretVolume = pod.getSpec.getVolumes.asScala.filter { volume =>
+      volume.getName == s"$TEST_SECRET_NAME-volume"
+    }
+    assert(testSecretVolume.size === 1)
+    assert(testSecretVolume.head.getSecret.getSecretName === TEST_SECRET_NAME)
+
+    checkTestSecretInContainer(pod.getSpec.getContainers.get(0))
+
+    if (withInitContainer) {
```

Comment:
Is this option being exercised currently?

Reply:
No, none of the tests uses an init-container.